tests/foreman/endtoend/test_api_endtoend.py

```diff
@@ -811,271 +890,190 @@ def test_positive_ping(self):
             content_view=content_view,
+            name=puppet_module['name'],
         ).create()
         self.assertEqual(
             puppet.name,
             puppet_module['name'],
         )
 
-        # step 2.9: Publish content view
+        # step 2.11: Publish content view
         content_view.publish()
 
-        # step 2.10: Promote content view to both lifecycles
+        # step 2.12: Promote content view to the lifecycle environment
         content_view = content_view.read()
         self.assertEqual(len(content_view.version), 1)
         cv_version = content_view.version[0].read()
         self.assertEqual(len(cv_version.environment), 1)
         promote(cv_version, le1.id)
-        # Check that content view exists in 2 lifecycles
-        content_view = content_view.read()
-        self.assertEqual(len(content_view.version), 1)
-        cv_version = cv_version.read()
-        self.assertEqual(len(cv_version.environment), 2)
-        promote(cv_version, le2.id)
-        # Check that content view exists in 2 lifecycles
+        # check that content view exists in lifecycle
         content_view = content_view.read()
         self.assertEqual(len(content_view.version), 1)
         cv_version = cv_version.read()
-        self.assertEqual(len(cv_version.environment), 3)
 
-        # BONUS: Create a content host and associate it with promoted
-        # content view and last lifecycle where it exists
-        content_host = entities.System(
-            server_config,
+        # step 2.13: Create a new activation key
+        activation_key_name = gen_string('alpha')
+        activation_key = entities.ActivationKey(
+            name=activation_key_name,
+            environment=le1,
+            organization=org,
             content_view=content_view,
-            environment=le2
         ).create()
-        # Check that content view matches what we passed
-        self.assertEqual(content_host.content_view.id, content_view.id)
-        # Check that lifecycle environment matches
-        self.assertEqual(content_host.environment.id, le2.id)
+        # step 2.14: Add the products to the activation key
+        for sub in entities.Subscription(organization=org).search():
+            if sub.read_json()['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
+                activation_key.add_subscriptions(data={
+                    'quantity': 1,
+                    'subscription_id': sub.id,
+                })
+                break
+        # step 2.14.1: Enable product content
+        activation_key.content_override(data={'content_override': {
+            u'content_label': u'rhel-6-server-rhev-agent-rpms',
+            u'value': u'1',
+        }})
 
-        # step 2.11: Create a new libvirt compute resource
+        # BONUS: Create a content host and associate it with promoted
+        # content view and last lifecycle where it exists
+        if not bz_bug_is_open(1318938):
+            content_host = entities.System(
```

Comment:
@oshtaier I have talked with @abalakh and he already have a PR to do this #3391. That said I will let the conversion to the Host entity be handled by that PR.